scala/scala_import.bzl

```diff
@@ -24,20 +24,27 @@ def _scala_import_impl(ctx):
         ctx.label,
         jars2labels,
     )  #last to override the label of the export compile jars to the current target
+
+    current_target_providers = [JavaInfo(
+        output_jar = jar,
+        compile_jar = jar,
+        exports = [target[JavaInfo] for target in ctx.attr.exports],
+        deps = [target[JavaInfo] for target in ctx.attr.deps],
+        runtime_deps = [target[JavaInfo] for target in ctx.attr.runtime_deps],
+        source_jar = ctx.file.srcjar,
+        neverlink = ctx.attr.neverlink
+    )
+    for jar in current_target_compile_jars]
+
+    if not current_target_providers:
```

Comment:
this is for cases where there are no current jars but only exports?

Reply:
I wonder if the below shouldn't happen in any case. Not sure.

Reply:
Yes, for the case with only exports. I added a TODO to figure out if merging the exports providers is always needed. IMO it's not, because the `JavaInfo`s created above specify the exports dependencies.